plugin/trino-hive/src/main/java/io/trino/plugin/hive/coercions/TimestampCoercer.java

```diff
@@ -0,0 +1,102 @@
+import io.trino.spi.type.LongTimestamp;
+import io.trino.spi.type.TimestampType;
+import io.trino.spi.type.VarcharType;
+
+import java.time.LocalDateTime;
+import java.time.chrono.IsoChronology;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeFormatterBuilder;
+import java.time.format.ResolverStyle;
+
+import static io.trino.spi.type.Timestamps.MICROSECONDS_PER_SECOND;
+import static io.trino.spi.type.Timestamps.NANOSECONDS_PER_MICROSECOND;
+import static io.trino.spi.type.Timestamps.PICOSECONDS_PER_NANOSECOND;
+import static io.trino.spi.type.Varchars.truncateToLength;
+import static java.lang.Math.floorDiv;
+import static java.lang.Math.floorMod;
+import static java.time.ZoneOffset.UTC;
+import static java.time.format.DateTimeFormatter.ISO_LOCAL_DATE;
+import static java.time.format.DateTimeFormatter.ISO_LOCAL_TIME;
+
+public final class TimestampCoercer
+{
+    private static final DateTimeFormatter LOCAL_DATE_TIME = new DateTimeFormatterBuilder()
+            .parseCaseInsensitive()
+            .append(ISO_LOCAL_DATE)
+            .appendLiteral(' ')
+            .append(ISO_LOCAL_TIME)
+            .toFormatter()
+            .withResolverStyle(ResolverStyle.STRICT)
+            .withChronology(IsoChronology.INSTANCE);
+
+    private TimestampCoercer() {}
+
+    public static class ShortTimestampToVarcharCoercer
+            extends TypeCoercer<TimestampType, VarcharType>
+    {
+        public ShortTimestampToVarcharCoercer(TimestampType fromType, VarcharType toType)
+        {
+            super(fromType, toType);
+        }
+
+        @Override
+        protected void applyCoercedValue(BlockBuilder blockBuilder, Block block, int position)
+        {
+            long epochMicros = fromType.getLong(block, position);
+            long epochSecond = Math.floorDiv(epochMicros, MICROSECONDS_PER_SECOND);
+            int nanoFraction = floorMod(epochMicros, MICROSECONDS_PER_SECOND) * NANOSECONDS_PER_MICROSECOND;
+            toType.writeSlice(
+                    blockBuilder,
+                    truncateToLength(
+                            Slices.utf8Slice(
+                                    LOCAL_DATE_TIME.format(LocalDateTime.ofEpochSecond(epochSecond, nanoFraction, UTC))),
+                            toType));
+        }
+    }
+
+    public static class LongTimestampToVarcharCoercer
+            extends TypeCoercer<TimestampType, VarcharType>
+    {
+        public LongTimestampToVarcharCoercer(TimestampType fromType, VarcharType toType)
+        {
+            super(fromType, toType);
+        }
+
+        @Override
+        protected void applyCoercedValue(BlockBuilder blockBuilder, Block block, int position)
+        {
+            LongTimestamp timestamp = (LongTimestamp) fromType.getObject(block, position);
+
+            long seconds = floorDiv(timestamp.getEpochMicros(), MICROSECONDS_PER_SECOND);
+            long microsFraction = floorMod(timestamp.getEpochMicros(), MICROSECONDS_PER_SECOND);
```

Comment:
inline (as above)

Reply:
Can you provide some context here.

Reply:
above you have
```
int nanoFraction = floorMod(epochMicros, MICROSECONDS_PER_SECOND) * NANOSECONDS_PER_MICROSECOND;
```

without `microsFraction` intermediate variable

why not write the same logic using same style?